### DISCUSSION TITLE
Add a Reset or Restart feature

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [mdgrs-mei/PowerShellRun#51] [Ideas]
If I navigate through a favorite folder but want to return to the "top"  I have to exit and restart it. Is there a way to reset that I'm missing? Otherwise, I'd love to have a keybinding that would restore the runner interface.

**Comment by mdgrs-mei** [comment]
There isn't any way to reset the interface at the moment.

If you are in a nested menu, you can go back to the previous menu with the `Backspace` key. Does that meet your requirement? or do you want to return to the top menu with a single step no matter how deep you are in the hierarchy?

**Comment by jdhitsolutions** [comment]
`Backspace` doesn't quite work. If I'm in a folder and backspace, it will take me to the parent folder. If I am in the parent folder, `Backspace` quits the app. It doesn't take me back to the initial run menu. Personally, I'd love a hotkey that would take me back to the initial run menu.

**Reply by mdgrs-mei** [reply]
ok, so you'd like a hotkey that just does the same as `quit` + `Invoke-PSRun`.

This might take some time to implement as there is not a suitable place to add this option currently (`KeyBinding` is for the selector but the nested menus are in the launcher). I'll add this as a new feature after some refactoring.

**Reply by jdhitsolutions** [reply]
Or find a way for `Backspace` to take you back to the initial state.

**Comment by jdhitsolutions** [comment]
If there was a way to map a keyboard macro that would be ideal.

**Comment by mdgrs-mei** [comment]
I'm working on this Restart feature in PR #76.  It will be included in the next release.

**Reply by mdgrs-mei** [reply]
@jdhitsolutions `$option.KeyBinding.RestartKeys` has been added in v0.9.0. It's mapped to `Ctrl+r` by default. It feels quite handy, thank you for your suggestions.

**Reply by jdhitsolutions** [reply]
This is handy. And `backspace` works better in v0.9.0 as well.

**Comment by jdhitsolutions** [comment]
Is there a command I can put in a scriptblock that would invoke the restart?

**Reply by mdgrs-mei** [reply]
No. If there is, what will you use it for?

**Reply by jdhitsolutions** [reply]
I have several scriptblocks where I want to run the code and then return to the PowerShellRun menu. Right now, I am running `Invoke-PSRun` at the end of my scriptblock. I'm wondering if using a command-equivalent of the new restart feature would be better. Maybe they are really the same thing.

**Reply by mdgrs-mei** [reply]
If you call `Invoke-PSRun` in your scriptblock, it is recursively called and the stack is one level deeper. The command-equivalent restart feature would be better as it's not a recursive call but it would not be that different unless you call the nested `Invoke-PSRun` thousands of times.

**Comment by jdhitsolutions** [comment]
It sounds like I wouldn't notice a performance difference, but a command equivalent to `ctrl+r` would be better programmatically.

**Reply by mdgrs-mei** [reply]
I would like to wait a bit until another requester comes up to see more use cases.
